builtin/provisioners/habitat/resource_provisioner.go

```diff
@@ -0,0 +1,745 @@
+package habitat
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/communicator"
+	"github.com/hashicorp/terraform/communicator/remote"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+	linereader "github.com/mitchellh/go-linereader"
+)
+
+const installURL = "https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.sh"
+const systemdUnit = `[Unit]
+Description=Habitat Supervisor
+
+[Service]
+ExecStart=/bin/hab sup run %s
```

Comment:
I think here we optionally need something like:
```
Environment="HAB_AUTH_TOKEN=<github_token>"
```
for anyone accessing private origins in Builder.